spec/puppetserver/ca/sign_action_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppetserver/ca/sign_action'
+require 'puppetserver/ca/cli'
+require 'puppetserver/ca/logger'
+
+RSpec.describe 'Puppetserver::Ca::SignAction' do
+  let(:err)    { StringIO.new }
+  let(:inform) { StringIO.new }
+  let(:logger) { Puppetserver::Ca::Logger.new(:info, inform, err) }
+  let(:action) { Puppetserver::Ca::SignAction.new(logger) }
+
+  describe 'validation' do
+    before do
+      allow_any_instance_of(Puppetserver::Ca::SignAction).to receive(:run).
+          and_raise("Validation succeeded when it should have failed")
+    end
+
+    it 'does not expect an argument' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign','--all','foo'],STDOUT, err)
+      expect(err.string).to match(/Error:.*Unknown arguments or flags:.*foo/m)
+      expect(exit_code).to eq(1)
+    end
+    it 'does not allow --all with a valid certname call' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign','--certname','foo','--all'],STDOUT,err)
+      expect(err.string).to include('--all and --certname cannot be used together')
+      expect(exit_code).to eq(1)
+    end
+    it 'errors without any arguments' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign'],STDOUT,err)
+      expect(err.string).to include('No arguments given')
+      expect(exit_code).to eq(1)
+    end
+    it 'errors without a certname' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign','--certname'],STDOUT,err)
+      expect(err.string).to include('A certname to sign is required')
+      expect(exit_code).to eq(1)
+    end
+    it 'does not allow --certname and --all' do
+      exit_code = Puppetserver::Ca::Cli.run(['sign','--certname','--all'],STDOUT,err)
+      expect(err.string).to match(/Cannot use --all.*HTTP API/)
+      expect(exit_code).to eq(1)
+    end
+    it 'works with a single cert' do
+      results, exit_code = action.parse(['--certname','foo'])
+      expect(results['certname']).to eq(['foo'])
+      expect(exit_code).to eq(nil)
+    end
+    it 'works with a comma separated list of certs' do
+      results, exit_code = action.parse(['--certname','foo,bar,baz'])
+      expect(results['certname']).to eq(['foo','bar','baz'])
+      expect(exit_code).to eq(nil)
+    end
+  end
+
+  describe 'error handling' do
+    let(:response)  { Struct.new(:code, :body) }
+    let(:success)   { response.new('204', nil) }
+    let(:not_found) { response.new('404', 'Not Found') }
+    let(:empty)     { response.new('404', '[]') }
+
+    it 'signs certs with successfull put' do
```

Comment:
I don't think "signs certs" is what the CLI is doing in response to a successful put. The put signed the certs, the CLI is... logging and exiting with zero, right?